server/models/dataset_helpers/image_metadata.py

```diff
@@ -271,7 +276,19 @@ def transform(cls, value):
             if value in ['', 'unknown']:
                 value = None
             else:
+                result = re.match(cls._formatRegex, value)
+                if not result:
+                    raise BadFieldTypeException(
+                        name=cls.name, fieldType='float with units (um, mm, or cm)', value=value)
+                value, units = result.groups()
                 value = cls._coerceFloat(value)
+
+                # Convert to mm
+                if units == 'um':
+                    value /= 100
```

Comment:
”um" is micrometers, and is 10^-6. "mm" is 10^-3. So the conversation factor is 1000.0.

Reply:
Let's use explicit decimals for both of these factors. In Python2 an `int` divided by an `int` is a truncated `int`. Even though we expect a `float` from `_coerceFloat`, there's no reason to have another implicit assumption.